### DISCUSSION TITLE
Component testing with vue and vite is not loading the viteConfig

### DISCUSSION THREAD

**Discussion by jeevananthank** [cypress-io/cypress#24794] [Questions and Help]
Component testing with vue and vite is not loading the viteConfig

Have recently update to Cypress 11 and have updated cypress.config.ts like below for component testing.

> component: {
        devServer: {
            framework: 'vue',
            bundler: 'vite',
            viteConfig: {
                ...viteConfig,
                server: {
                    https: false,
                },
            },
        },
        supportFile: 'tests/unit/cypress/support/component.ts',
        indexHtmlFile: 'tests/unit/cypress/support/component-index.html',
        specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
    },

But upon running the tests, the alias defined in viteConfig is not loaded and the component tests fails with message 
*failed to resolve import "@/style/global.scss"*

**Comment by portyu9** [comment]
Hi @jeevananthank 

The error

```text
failed to resolve import "@/style/global.scss"
```

means Vite does not have the `@` alias that your application build has.

With current Cypress, the simplest Vue + Vite setup is normally to **remove the custom `viteConfig` entirely** and let Cypress discover and merge your project's `vite.config.*` automatically:

```js
component: {
  devServer: {
    framework: 'vue',
    bundler: 'vite',
  },
}
```

If you do need to pass it explicitly, make sure you're passing the actual Vite configuration object, including `resolve.alias`:

```js
import appViteConfig from './vite.config'

export default defineConfig({
  component: {
    devServer: {
      framework: 'vue',
      bundler: 'vite',
      viteConfig: appViteConfig,
    },
  },
})
```

Or declare the alias explicitly:

```js
viteConfig: {
  resolve: {
    alias: {
      '@': fileURLToPath(new URL('./src', import.meta.url)),
    },
  },
}
```

If `@` only exists in `tsconfig.json`/`jsconfig.json`, remember that Vite does not automatically turn TypeScript path aliases into Vite aliases; use the corresponding Vite configuration/plugin.

So I would first inspect the resolved `resolve.alias` rather than assuming Cypress is ignoring `viteConfig`.

Please mark this answer as accepted if it helped, thank you.
